nav2_system_tests/src/localization/CMakeLists.txt

```diff
@@ -15,4 +15,5 @@ ament_add_test(test_localization
     TEST_EXECUTABLE=$<TARGET_FILE:test_localization_node>
     TEST_WORLD=${PROJECT_SOURCE_DIR}/worlds/turtlebot3_ros2_demo.world
     GAZEBO_MODEL_PATH=${PROJECT_SOURCE_DIR}/models
+    RMW_IMPLEMENTATION=rmw_opensplice_cpp
```

Comment:
That's odd. You say setting the variable in the dockerfile didn't work

``` Dockerfile
ENV RMW_IMPLEMENTATION=rmw_opensplice_cpp
```

But setting it in the CMakeList does?
Is this specific to tests, or includes running anything manually in container's shell?

Reply:
> not in the Dockerfile apparently.

Oh, I'll try this out tomorrow when I get some time. Although, I think you'd want to set these kind of  parameters, like what rmw implementation to use, in the CI configs so that they can be matrix'ed.

https://github.com/ros-planning/navigation2/blob/178c7c6ce4aa3ea58189302e1287d44f800543cd/.circleci/config.yml#L205-L207

https://github.com/ros-planning/navigation2/blob/178c7c6ce4aa3ea58189302e1287d44f800543cd/.travis.yml#L27-L29